### ISSUE TITLE
MetadataRetrievalMethod=ProxyAPI sending wrong catalog in request to get table metadata

### ISSUE THREAD

**Issue by sfjkastello** [dbeaver/dbeaver#18786]
### Description

Issue with getting tables in one table when accessing Athena with MetadataRetrievalMethod=ProxyAPI;
Addition use can remove MetadataRetrievalMethod=ProxyAPI;  they then can see the tables in Account A, but recieves similar glue:GetTables error when running select against a table in Account A that was not displaying when MetadataRetrievalMethod=ProxyAPI; was in the connection string.

Case with AWS resulted in them being able to re-create the issue, and they have determined this is an issue specific to 
It appears that DBeaver is not handling well the situation when there are multiple data sources and sends the wrong catalog in the request to get table metadata.

### DBeaver Version

Community Edition 22.3.1

### Operating System

Microsoft Windows 10

### Database and driver

Database: Athena/Glue
Driver:  2.0.34.10000

### Steps to reproduce

1. Athena Driver has MetadataRetrievalMethod=ProxyAPI set in jdbc connection.
2. Have a role with the ability to connect to multiple databases in multiple accounts.
4. Connect to Athena using that role.
5. See all databases in Account A and Account B
5. Try to expand tables in all of the databases.  
6. On one table (in our situation, first Database in Account A), we receive the error that we do not have access glue:GetTables on a database in Account B... but the database only exists in Account A.

SQL Error [100071] [HY000]: [Simba][AthenaJDBC](100071) An error has been thrown from the AWS Athena client. is not authorized to perform: glue:GetTables
User: arn:aws:sts::Account A:assumed-role/authorizedrole/authorizeduser  is not authorized to perform: glue:GetTables on resource: arn:aws:glue:region:Account B:database/xxxxxx 
because no resource-based policy allows the glue:GetTables action (Service: AmazonDataCatalog; Status Code: 400; Error Code: AccessDeniedException; 
Request ID: xxxx-xxxx-xxxx-xxxx-xxxxx; Proxy: null) [Execution ID not available]

**Comment by LonwoLonwo** [comment]
Hello @sfjkastello 

Could you please open the Query manager and enable all filters? The instruction is here: https://github.com/dbeaver/dbeaver/wiki/Query-Manager
And then do screens about table metadata reading from the Query Manager?
Like this:
![2023-01-31 11_44_05-DBeaver Ultimate 22 3 3 - 11 24 324FFd](https://user-images.githubusercontent.com/45152336/215711468-60563d9a-50c5-4742-a601-f7c42d04a059.png)

**Comment by sfjkastello** [comment]
Here is the screen shot of the query manager... Actions taken:
1. Opened the connection
2. Expanded the connection
3. Tried to expand the tables for the first database listed ... got the error originally sent... nothing is listed in the query manager.
![QueryManger](https://user-images.githubusercontent.com/67717572/215776586-c2c38650-fdbf-4364-96b3-a71e8e345789.jpg)

If I try to expand the the tables in another database it will put the entry like this in the query manger:
-- Load tables [null, default, %, null]

**Comment by LonwoLonwo** [comment]
Could you please provide the log files? You can find instructions here: https://dbeaver.com/docs/wiki/Log-files/. Thank you.

**Comment by sfjkastello** [comment]
Attached is the debug log... scrubbed

[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/10547537/dbeaver-debug.log)

**Comment by LonwoLonwo** [comment]
```java
2023-01-31 07:34:48.834 - User: arn:aws:sts::Account A:assumed-role/authorizedrole/authorizeduser is not authorized to perform: glue:GetTables on resource: arn:aws:glue:us-east-1:Account B:database/xxxxxxx because no resource-based policy allows the glue:GetTables action (Service: AmazonDataCatalog; Status Code: 400; Error Code: AccessDeniedException; Request ID: a5685b5c-c4da-4efc-b447-5c174346dd60; Proxy: null) (Service: AmazonAthena; Status Code: 400; Error Code: MetadataException; Request ID: e2a21ae7-634c-48a9-9a46-53603fa509fb; Proxy: xxxx.xxxx.xxxx.xxxx)
com.simba.athena.amazonaws.services.athena.model.MetadataException: User: arn:aws:sts::Account A:assumed-role/authorizedrole/authorizeduser is not authorized to perform: glue:GetTables on resource: arn:aws:glue:us-east-1:Account B:database/xxxxxxx because no resource-based policy allows the glue:GetTables action (Service: AmazonDataCatalog; Status Code: 400; Error Code: AccessDeniedException; Request ID: a5685b5c-c4da-4efc-b447-5c174346dd60; Proxy: null) (Service: AmazonAthena; Status Code: 400; Error Code: MetadataException; Request ID: e2a21ae7-634c-48a9-9a46-53603fa509fb; Proxy: xxxx.xxxx.xxxx.xxxx)
	at com.simba.athena.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1879)
	at com.simba.athena.amazonaws.http.AmazonHttpClient$RequestExecutor.handleServiceErrorResponse(AmazonHttpClient.java:1418)
	at com.simba.athena.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1387)
	at com.simba.athena.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1157)
	at com.simba.athena.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:814)
	at com.simba.athena.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:781)
	at com.simba.athena.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:755)
	at com.simba.athena.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:715)
	at com.simba.athena.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:697)
	at com.simba.athena.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:561)
	at com.simba.athena.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:541)
	at com.simba.athena.amazonaws.services.athena.AmazonAthenaClient.doInvoke(AmazonAthenaClient.java:2570)
	at com.simba.athena.amazonaws.services.athena.AmazonAthenaClient.invoke(AmazonAthenaClient.java:2537)
	at com.simba.athena.amazonaws.services.athena.AmazonAthenaClient.invoke(AmazonAthenaClient.java:2526)
	at com.simba.athena.amazonaws.services.athena.AmazonAthenaClient.executeListTableMetadata(AmazonAthenaClient.java:1862)
	at com.simba.athena.amazonaws.services.athena.AmazonAthenaClient.listTableMetadata(AmazonAthenaClient.java:1831)
	at com.simba.athena.athena.api.AJClient.fetchTablesWithProxyApi(Unknown Source)
	at com.simba.athena.athena.api.AJClient.getTablesMetadata(Unknown Source)
	at com.simba.athena.athena.dataengine.metadata.AJTablesMetadataSource.initMetadata(Unknown Source)
	at com.simba.athena.athena.dataengine.metadata.AJTablesMetadataSource.<init>(Unknown Source)
	at com.simba.athena.athena.dataengine.AJDataEngine.makeNewMetadataSource(Unknown Source)
	at com.simba.athena.dsi.dataengine.impl.DSIDataEngine.makeNewMetadataResult(Unknown Source)
	at com.simba.athena.athena.dataengine.AJDataEngine.makeNewMetadataResult(Unknown Source)
	at com.simba.athena.jdbc.jdbc42.S42DatabaseMetaData.createMetaDataResult(Unknown Source)
	at com.simba.athena.jdbc.common.BaseDatabaseMetaData.getTables(Unknown Source)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCDatabaseMetaDataImpl.getTables(JDBCDatabaseMetaDataImpl.java:1116)
	at org.jkiss.dbeaver.ext.generic.model.meta.GenericMetaModel.prepareTableLoadStatement(GenericMetaModel.java:585)
```

> user is not authorized to perform: glue:GetTables

Ok. Are you sure, that you have enough permissions for this action?

**Comment by sfjkastello** [comment]
The issue seems to be that the catalog being sent with the request in isn't correct.  The database in Account B doesn't exist.  The database only exists in account A.  There are no permissions to provide to a non-existent database.

**Comment by sfjkastello** [comment]
Not sure if last comment was oriented toward OP... but... not sure what that question would mean.  Seems it is related to the first database listed.  When opening the connection in Database Navigator, then expanding database, when trying to expand the tables we get the error listed.  Even though the database is only available in Account A... it is throwing an error saying there is no access to the database in account B.

**Comment by E1izabeth** [comment]
Thank you for update.
We're waiting for the test environment to reproduce your issue. It will take some time.

**Comment by sfjkastello** [comment]
Additional information... we tried to catalog a dummy database out there 'adummy' that has no backend hoping to trick it into skipping the 'first'.  When connecting to this we got the same permissions error pointing to Account B... and the original database that was giving the error still errored the same error.

**Comment by LonwoLonwo** [comment]
Hello @sfjkastello 

Could you please provide a screen from your database navigator tree? I still can't reproduce the issue. With provided steps, I have normal different tables structure in different schemas.

Also, do you have special Driver Properties in your connection settings? I can not use "MetadataRetrievalMethod=ProxyAPI", it gives me an error.

**Comment by sfjkastello** [comment]
Attached is the tree view.  We tried to create a dummy database (adummy) to see if we could trick the seemingly "first database issue", however we get the same error with that database as well.  So, the first two databases are throwing this error, but not the subsequent ones.

I am using: com.simba.athena.amazonaws.auth.profile.ProfileCredentialsProvider as my AWSCredentialProvider.

My URL template in my driver manager is:
jdbc:awsathena://AWSRegion=us-east-1;MetadataRetrievalMethod=ProxyAPI;

Just upgraded JAR to AthenaJDBC42-2.0.36.1000.jar in case make any difference (it didn't): https://docs.aws.amazon.com/athena/latest/ug/connect-with-jdbc.html

When I take the MetadataRetrievalMethod=ProxyAPI; out I can expand the database, however, I cannot see any databases in any other account.

![Athena_Tree](https://github.com/dbeaver/dbeaver/assets/67717572/ed808d96-213a-4e6e-b142-d2f18bab76ac)

Using: DBeaver Version 22.3.1.202212251901

**Comment by LonwoLonwo** [comment]
Finally! We changed our test Athena environment and can see different data sources in the Navigator tree. 
But we need to change our code for Athena and add a special setting like "Show data sources" on the Athena Connection Page.

**Comment by serge-rider** [comment]
This feature can be enabled by turning on "Show catalogs" option.
Will be available in 23.1.4

![image](https://github.com/dbeaver/dbeaver/assets/6398845/2509e3bf-176b-48c8-a5c0-8cf6296e91b1)

**Comment by sfjkastello** [comment]
GREAT NEWS!!!  Very much looking forward to it!!!  Thank you!
